api/src/plugins/access-token-scheme.js

```diff
@@ -58,6 +59,21 @@ exports.register = function (server, pluginOptions, next) {
         provider[TokenModel].find(tokenString).then(token => {
           if (token) {
             token.scope = token.scope.split(' ');
+
+            if (includes(token.scope, 'superadmin')) {
+              const provider = server.plugins['open-id-connect'].provider;
+              provider.Client.find(token.clientId).then(client => {
+
+                if (!client) {
+                  return reply(Boom.notFound('Client not found'));
+                }
+
+                if (!client.superadmin) {
+                  reply(Boom.forbidden('invalid superadmin scope'))
```

Comment:
Have this `return` here, otherwise code execution will continue when maybe it shouldn't. Also missing a semicolon.